### ISSUE TITLE
Flashlight-Monitoring (via MBeans) not working in payara embedded

### ISSUE THREAD

**Issue by aojava** [payara/Payara#818]
While Monitoring in `payara full` works as expected, but in payara `embedded all` it doesn't. Which information should I provide to help you finding the problem?

Stacktrace:

```
16.05.16 17:50:28 FEIN         : registering a statsProvider                                                                                                                                          [Mo, 16.05.2016 17:50:28.292 MESZ]: org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl tryToRegister registering a statsProvider  [javax.enterprise.monitoring]
16.05.16 17:50:28 FEIN         :  enabled is true                                                                                                                                                     [Mo, 16.05.2016 17:50:28.293 MESZ]: org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl tryToRegister  enabled is true   [javax.enterprise.monitoring]
16.05.16 17:50:28 FEIN         : Enabling the statsProvider - com.sun.enterprise.v3.services.impl.monitor.stats.ThreadPoolStatsProvider                                                               [Mo, 16.05.2016 17:50:28.294 MESZ]: org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl enableStatsProvider Enabling the statsProvider - com.sun.enterprise.v3.services.impl.monitor.stats.ThreadPoolStatsProvider  [javax.enterprise.monitoring]
16.05.16 17:50:28 WARNUNG      : Flashlight listener registration failed for listener class: com.sun.enterprise.v3.services.impl.monitor.stats.ThreadPoolStatsProvider , will retry later             [Mo, 16.05.2016 17:50:28.298 MESZ]: org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl register Flashlight listener registration failed for listener class: com.sun.enterprise.v3.services.impl.monitor.stats.ThreadPoolStatsProvider , will retry later  [javax.enterprise.monitoring]
16.05.16 17:50:28 FEIN         : Listener registration failed                                                                                                                                         [Mo, 16.05.2016 17:50:28.298 MESZ]: org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl register Listener registration failed 
java.lang.RuntimeException: Probe is not yet registered: glassfish:kernel:thread-pool:threadDispatchedFromPoolEvent
    at org.glassfish.flashlight.impl.client.FlashlightProbeClientMediator.handleListenerAnnotations(FlashlightProbeClientMediator.java:245)
    at org.glassfish.flashlight.impl.client.FlashlightProbeClientMediator.registerJavaListener(FlashlightProbeClientMediator.java:145)
    at org.glassfish.flashlight.impl.client.FlashlightProbeClientMediator.registerListener(FlashlightProbeClientMediator.java:122)
    at org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl.registerStatsProviderToFlashlight(StatsProviderManagerDelegateImpl.java:681)
    at org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl.enableStatsProvider(StatsProviderManagerDelegateImpl.java:431)
    at org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl.tryToRegister(StatsProviderManagerDelegateImpl.java:202)
    at org.glassfish.admin.monitor.StatsProviderManagerDelegateImpl.register(StatsProviderManagerDelegateImpl.java:161)
    at org.glassfish.external.probe.provider.StatsProviderManager.registerStatsProvider(StatsProviderManager.java:91)
    at org.glassfish.external.probe.provider.StatsProviderManager.register(StatsProviderManager.java:66)
    at org.glassfish.external.probe.provider.StatsProviderManager.register(StatsProviderManager.java:57)
    at com.sun.enterprise.v3.services.impl.monitor.GrizzlyMonitoring.registerThreadPoolStatsProvider(GrizzlyMonitoring.java:148)
    at com.sun.enterprise.v3.services.impl.GlassfishNetworkListener.registerMonitoringStatsProviders(GlassfishNetworkListener.java:290)
    at com.sun.enterprise.v3.services.impl.GlassfishNetworkListener.configureHttpProtocol(GlassfishNetworkListener.java:159)
    at org.glassfish.grizzly.config.GenericGrizzlyListener.configureSubProtocol(GenericGrizzlyListener.java:409)
    at org.glassfish.grizzly.config.GenericGrizzlyListener.configureProtocol(GenericGrizzlyListener.java:398)
    at org.glassfish.grizzly.config.GenericGrizzlyListener.configure(GenericGrizzlyListener.java:294)
    at com.sun.enterprise.v3.services.impl.GrizzlyProxy.initialize(GrizzlyProxy.java:124)
    at com.sun.enterprise.v3.services.impl.GrizzlyService.createNetworkProxy(GrizzlyService.java:539)
    at com.sun.enterprise.v3.services.impl.GrizzlyService.postConstruct(GrizzlyService.java:490)
    at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:329)
    at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:377)
    at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:471)
    at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:227)
    at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:84)
    at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2064)
    at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:105)
    at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:87)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.oneJob(CurrentTaskFuture.java:1212)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.run(CurrentTaskFuture.java:1143)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpOneLevel.run(CurrentTaskFuture.java:761)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.go(CurrentTaskFuture.java:560)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.access$100(CurrentTaskFuture.java:442)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:153)
    at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:148)
    at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.proceedTo(AsyncRunLevelContext.java:435)
    at org.glassfish.hk2.runlevel.internal.RunLevelControllerImpl.proceedTo(RunLevelControllerImpl.java:70)
    at com.sun.enterprise.v3.server.AppServerStartup.proceedTo(AppServerStartup.java:534)
    at com.sun.enterprise.v3.server.AppServerStartup.run(AppServerStartup.java:331)
    at com.sun.enterprise.v3.server.AppServerStartup.doStart(AppServerStartup.java:228)
    at com.sun.enterprise.v3.server.AppServerStartup.start(AppServerStartup.java:219)
    at com.sun.enterprise.glassfish.bootstrap.GlassFishImpl.start(GlassFishImpl.java:79)
    at de.aidaorga.aida.ewas.starter.Ewas.start(Ewas.java:407)
    at de.aidaorga.aida.ewas.starter.JniWrapper$1.run(JniWrapper.java:40)
    at java.lang.Thread.run(Thread.java:745)
 [javax.enterprise.monitoring]
```

**Comment by fturizo** [comment]
Hi @aojava. How are you setting the monitoring for the Payara embedded all in your application? We'd appreciate if you can provide a code snippet to help replicate this bug.

**Comment by aojava** [comment]
In the attachement, you'll find a small demo project. Does this help?
[starter.zip](https://github.com/payara/Payara/files/331752/starter.zip)

**Comment by fturizo** [comment]
Hi @aojava, many thanks for your code sample. I could reproduce your issue and got the same exceptions for the Flashlight monitor registering for several components of the server. In particular, after tinkering with the monitoring configuration for a while, I noticed that these exceptions appeared only when I set the configuration level to **HIGH** for the following components:

-   thread-pool
-   http-service
-   transaction-service
-   web-container
-   deployment

I raised an issue for our developer team to investigate a possible fix to this bug. We'll update this issue when we have more information.

**Comment by mikecroft** [comment]
Internal issue ID `PAYARA-855`

**Comment by smillidge** [comment]
Please try Payara 5 and see if it is still a problem there.

**Comment by smillidge** [comment]
Closed a long time ago by #1643
